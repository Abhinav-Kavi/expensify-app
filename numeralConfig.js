import numeral from 'numeral';

const numeralConfig = (() => {
  numeral.register('locale', 'In', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '₹'
    }
  });

  // switch between locales
  numeral.locale('In');
})();