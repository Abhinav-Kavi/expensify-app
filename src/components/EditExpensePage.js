import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {

  onSubmit = (expenseUpdates) => {
    const id = this.props.expense.id;
    this.props.editExpense(id, expenseUpdates);
    this.props.history.push('/');
  }

  onClick = (e)=>{
    this.props.removeExpense({id : this.props.expense.id});
    this.props.history.push('/');
  }

  render(){
    return(
      <div>
        <ExpenseForm 
        expense = {this.props.expense}
        onSubmit = {this.onSubmit}
       />
       <button 
          onClick = {this.onClick}
        >
          Remove
        </button>
    </div>
    );
  }
}

const mapStateToProps = (state,props) => ({
  expense : state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch,props) => ({
  editExpense : (id,expenseUpdates) => dispatch(editExpense(id,expenseUpdates)),
  removeExpense: (data) => dispatch(removeExpense(data))
});

export default  connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);