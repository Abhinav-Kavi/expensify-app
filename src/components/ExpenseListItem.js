import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({description,amount,createdAt,id})=>(
  <div>
    <Link to={`/edit/${id}`}> <h3>{description}</h3> </Link>
    <p>
     {numeralConfig()}
     {numeral(amount).format('$0,0.00')} 
     -
     {moment(createdAt).format('MMMM Do, YYYY')}</p>   
  </div>
);

const numeralConfig = () => {
  numeral.register('locale', 'In', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '₹'
    }
  });

  // switch between locales
  numeral.locale('In');
}

export default ExpenseListItem;