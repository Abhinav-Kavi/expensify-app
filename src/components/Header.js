import React from 'react';
import {NavLink} from 'react-router-dom';

const Header =()=>(
  <div>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" to="/" exact={true}>Dashboard</NavLink>
    <NavLink activeClassName="is-active" to="/create">Create</NavLink>

  </div>
);

export default Header;