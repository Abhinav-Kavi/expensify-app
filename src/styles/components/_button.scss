.button {
  background: $blue;
  color: white;
  display: inline-block;
  border: none;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--secondary{
  background: #888;
  margin-bottom: $m-size;
}

.button--link {
  background: none;
}
