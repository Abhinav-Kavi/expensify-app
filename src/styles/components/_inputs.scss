.text-input {
  border : 1px solid #cacccd;
  font-size: $font-size-large;
  font-weight: 300;
  height: 48px;
  padding: $s-size;
}


.textarea {
  @extend .text-input;
  height: 10rem;
}


.select {
  @extend .text-input;
}
