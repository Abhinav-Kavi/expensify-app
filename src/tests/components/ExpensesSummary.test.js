import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import '../../numeralConfig';

test('should render ExpensesSummary correctly with 1 expense',()=>{
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={100}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses',()=>{
  const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={1023.34}/>);
  expect(wrapper).toMatchSnapshot();
});