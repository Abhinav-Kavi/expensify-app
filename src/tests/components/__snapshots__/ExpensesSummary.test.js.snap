// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    ₹100.00
    .
  </h2>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    5
     
    expenses
     totalling 
    ₹1,023.34
    .
  </h2>
</div>
`;
